Fix: Rebuilt Text Tool to Match Excalidraw UX and Fix Sync Issues

f0rsakeN-afk opened the pull request:
This PR redesigns the Text Tool in CollabyDraw to match the UX of Excalidraw.

✅ Fixes:
- Textarea placement bugs
- Styling and font inconsistencies
- Positioning glitches

Closes #7 

https://github.com/user-attachments/assets/53a67de4-1a61-4095-b094-a8970359f7df

coderomm commented:
@f0rsakeN-afk 

### I want to show this points as well.
1. How its looking when pressing enter and how you handling shift + enter for entering into newline.
2. How its looking when we type something and then scoll through the page.

f0rsakeN-afk commented:
I’ve pushed the requested fixes addressing the review comments, including the textarea enter/shift-enter behavior.
Please take another look when you get a chance. Thanks!

https://github.com/user-attachments/assets/07b9cef9-34a9-48f7-b435-509af5975f49

Closes #7
